File: SMSfactvocab/SMSfactvocab.R
# clear workspace
 graphics.off()
 rm(list=ls(all=TRUE))

# setwd("C:/...") #set your working directory
load("vocabulary.rda")

voc       = vocabulary[,1:4] 
factvocab = factanal(voc, factors=1,scores="Bartlett")
facttable = cbind(factvocab$loadings, (1 - factvocab$uniquenesses),factvocab$uniqueness)
colnames(facttable) = c("q1", "Communalities", "Specific variances")
print(round(facttable,4))
# xtable(facttable,digits=4)
plot(factanal(voc, factors=1,scores="Bartlett")$scores,type="n",xlab="",ylab="Score")
text(factanal(voc, factors=1,scores="Bartlett")$scores,rownames(voc))

File: SMSfactvocab/SMSfactvocab.m
clear all
close all
clc

x         = load('vocabulary.dat');
voc       = x(:,1:4); 
[lambda,psi,T,stats,F] = factoran(voc, 1, 'scores', 'Bartlett')
facttable = [lambda, (1 - psi), psi];
format bank
disp('          q1            Communalities Specific variances')
disp(facttable)
format short
lab = strvcat(num2str((1:64)'));
plot((1:64)',F,'w')
text((1:64)',F(:),lab,'FontSize',14)
ylabel('Score','FontSize',16,'FontWeight','Bold')
set(gca,'LineWidth',1.6,'FontSize',16,'FontWeight','Bold')